include/onnxruntime/core/common/common.h

```diff
@@ -95,6 +99,57 @@ void LogRuntimeError(uint32_t session_id, const common::Status& status, const ch
 #define ORT_WHERE_WITH_STACK \
   ::onnxruntime::CodeLocation(__FILE__, __LINE__, __PRETTY_FUNCTION__, ::onnxruntime::GetStackTrace())
 
+#ifdef ORT_NO_EXCEPTIONS
+
+#define ORT_TRY
+
```

Comment:
Should we have an ORT_CATCH?  

Possibly something like [https://www.boost.org/doc/libs/1_74_0/boost/core/no_exceptions_support.hpp](https://www.boost.org/doc/libs/1_74_0/boost/core/no_exceptions_support.hpp) would work. Not sure if you really need the 'try' to create a new scope (if not the BOOST_CATCH_END wouldn't be required).

I started with ORT_CATCH and the Eigen approach, but that didn't work for multiple catch statements. The Boost approach should be more robust and having ORT_CATCH would make the usage in the code a bit less ugly (although painful to update the PR - sorry!). Quick test is probably converting a try/catch with a std::exception catch and a catch-all 'catch(...)' like the one in inference_session.cc. If that works you should be fine to adopt this approach everywhere.